### ISSUE TITLE
Qutebrowser as the pulseaudio sink name

### ISSUE THREAD

**Issue by TomRiddle01** [qutebrowser/qutebrowser#3832]
Currentlty the application.name and sink name of qutebrowser (QtWebengine) are called "Chromium". 
This can be seen in either pavucontrol, pulsemixer, pactl or pacmd:

```
» pacmd list-sink-inputs | grep Chromium
    client: 311 <Chromium>
        application.name = "Chromium"
        module-stream-restore.id = "sink-input-by-application-name:Chromium"
```

It would be nice to be able to differ between a qutebrowser instance and a real Chromium instance. In my specific example I'd like to direct Chromium and qutebrowser to two different outputs. Currently it remembers the last settings for "Chromium"

Would that technically be possible?

**Comment by The-Compiler** [comment]
I'm not sure - it's likely that qutebrowser can't do anything about it, but QtWebEngine could.

**Comment by toofar** [comment]
Allrighty, this can be changed by setting [AudioManager::SetGlobalAppName()](https://codereview.chromium.org/1317033006/) in the chromium source. I grepped through the qtwebengine source (as of a few months ago) and didn't see any usage of the method in chromium or webengine. So this appears to be an issue to be raised upstream . I assume they have the application name in QApplication somewhere. The only mention of it I found on the bug tracker is [this issue](https://bugreports.qt.io/browse/QTBUG-52124) which seems unrelated and triggered by no longer existing chromium code.

**Comment by The-Compiler** [comment]
I opened [QTBUG-67707](https://bugreports.qt.io/browse/QTBUG-67707).

**Comment by The-Compiler** [comment]
Looks like there's a fix in review for Qt 5.11 already (that was quick!): https://codereview.qt-project.org/#/c/226343/

**Comment by TomRiddle01** [comment]
Aweseome, amazingly fast!

**Comment by gslav** [comment]
**Version info**:
```qutebrowser v1.12.0
Git commit: 
Backend: QtWebEngine (Chromium 80.0.3987.163)
Qt: 5.15.0

CPython: 3.8.3
PyQt: 5.15.0

sip: 5.3.0
colorama: 0.4.3
pypeg2: 2.15
jinja2: 2.10.3
pygments: 2.6.1
yaml: 5.3.1
cssutils: no
attr: 19.3.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.0
PyQt5.QtWebKitWidgets: yes
pdf.js: no
sqlite: 3.32.2
QtNetwork SSL: OpenSSL 1.1.1g  21 Apr 2020

Style: Qt5CTProxyStyle
Platform plugin: xcb
OpenGL: Intel Open Source Technology Center, 3.0 Mesa 20.0.7
Platform: Linux-5.6.16-1-MANJARO-x86_64-with-glibc2.2.5, 64bit
Linux distribution: Manjaro Linux (manjaro)
Frozen: False
Imported from /usr/lib/python3.8/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt
```
**Does the bug happen if you start with `--temp-basedir`?**:
yes

**Description**
Again the application.name and sink name of qutebrowser (QtWebengine) are called "Chromium" and not "qutebrowser".
Can be solved on qutebrowser side? 

`pacmd list-sink-inputs` : 

```    index: 108
	driver: <protocol-native.c>
	flags: START_CORKED 
	state: RUNNING
	sink: 8 <jack_out.2>
	volume: front-left: 78862 / 120% / 4,82 dB,   front-right: 78862 / 120% / 4,82 dB
	        balance 0,00
	muted: no
	current latency: 74,46 ms
	requested latency: 42,67 ms
	sample spec: float32le 2-канальный 
	channel map: front-left,front-right
	             Стерео
	resample method: speex-float-1
	module: 11
	client: 224 <Chromium>
	properties:
		application.icon_name = "chromium-browser"
		media.name = "Playback"
		application.name = "Chromium"
		native-protocol.peer = "UNIX socket client"
		native-protocol.version = "33"
		application.process.id = "1721031"
		application.process.user = "g"
		application.process.host = "u"
		application.process.binary = "QtWebEngineProcess"
		application.language = "ru_RU.utf8"
		window.x11.display = ":0"
		application.process.machine_id = "bbb1ce18cd3f48f5b8af938475c2041c"
		application.process.session_id = "2"
		module-stream-restore.id = "sink-input-by-application-name:Chromium"
```

**Comment by The-Compiler** [comment]
@gslav Nothing qutebrowser can do, as far as I know - but I opened [[QTBUG-85363] [REG 5.13 -> 5.14] QtWebEngine is listed as "Chromium" in PulseAudio - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-85363).

**Comment by The-Compiler** [comment]
Turns out I was wrong, there are `PULSE_PROP_OVERRIDE_*` environment variables which can be set by qutebrowser and will be inherited by the `QWebEngineProcess` subprocesses actually playing the audio. This is now done in 7d388163106b4efa07b13ee515e0534a43cd6510.

**Comment by rien333** [comment]
Neat!

On a vaguely related note, is there anyway to rename the media player name as shown by MPRIS? `playerctl` still lists it as chromium (I hoped that would've changed with this commit)

**Comment by The-Compiler** [comment]
@rien333 Looks like that one is hardcoded in Chromium :worried: https://source.chromium.org/chromium/chromium/src/+/master:components/system_media_controls/linux/system_media_controls_linux.cc;l=162-166;drc=e31f2d43649586f3fb71f52098f66bb1c19d1b8a

**Comment by The-Compiler** [comment]
@rien333 FWIW I opened [[QTBUG-85475] Use media::AudioManager::GetGlobalAppName for MPRIS integration - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-85475)

**Comment by The-Compiler** [comment]
@rien333 Looks like a fix will probably make it into Qt 5.15.1: [Use audio-manager app-name for MPRIS (I8600fa86) · Gerrit Code Review](https://codereview.qt-project.org/c/qt/qtwebengine-chromium/+/307217)

**Comment by The-Compiler** [comment]
This doesn't seem to work with Qt 6 anymore, see #7431
